sympy/core/tests/test_expr.py

```diff
@@ -1671,8 +1671,9 @@ def test_round():
     r = Float(str(n)).round(-4)
     assert r == 10000
     # in fact, it should equal many values since __eq__
-    # compares at equal precision
-    assert all(r == i for i in range(9984, 10049))
+    # compares at equal precision (but it knows that
+    # the answer is not odd)
+    assert all(r == i for i in range(9984, 10049, 2))
```

Comment:
@jksuom, I now see that the way the comparison is made already handles the "interval nature" of the Float by doing the comparison of the `Float` with `Float(Rational)` which forces the Rational either into the same bin as the Float or not. For this reason, there may be many Rationals that will compare as being equal to a Float as this tests shows. Given this fuzziness, is it a good idea to allow for parity discrimination? Notice how the Float `r` compares equal to many *even* values but no *odd* values.

Reply:
> there may be many Rationals that will compare as being equal to a Float

It would follow from transitivity of == that those Rationals would also be equal. However, I think that transitivity is not implemented in SymPy, so that may not be an issue. But still, that does not feel right. It would be interesting to see the consequences of Rational == Float being always False. None would probably behave similarly (unless forbidden somewhere) but it is harder to implement. Making it False seems to me analogous to `x == 2` being False in SymPy although `x` could take the value 2.